src/State/ScreenProcessor.php

```diff
@@ -44,6 +56,119 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
+                $region = $this->screenLayoutRegionsRepository->findOneBy(['id' => $regionAndPlaylists['regionId']]);
+
+                if (is_null($region)) {
+                    throw new InvalidArgumentException('Unknown region resource');
+                }
+
+                $newPlaylists = array_map(
+                    /**
+                     * @param array<mixed> $playlistObject
+                     *
+                     * @return Ulid
+                     */
+                    fn ($playlistObject): Ulid => Ulid::fromString($playlistObject['id']), $regionAndPlaylists['playlists']);
+
+                $playlistScreens = $this->playlistScreenRegionRepository->findBy([
+                    'screen' => $screen->getId(),
+                    'region' => $regionAndPlaylists['regionId'],
+                ]);
+
+                $existingPlaylists = array_map(function ($playlistObject) {
+                    $playlist = $playlistObject->getPlaylist();
+                    if (!is_null($playlist)) {
+                        return $playlist->getId();
+                    }
+                }, $playlistScreens);
+
+                // This diff finds the playlists to be deleted
+                $deletePlaylists = array_diff($existingPlaylists, $newPlaylists);
+
+                // ... and deletes them.
+                foreach ($deletePlaylists as $deletePlaylist) {
+                    $regionId = $region->getId();
+                    $screenId = $screen->getId();
+                    if (!is_null($screenId) && !is_null($regionId) && !is_null($deletePlaylist)) {
+                        $this->playlistScreenRegionRepository->deleteRelations($screenId, $regionId, $deletePlaylist);
+                    }
+                }
+
+                // This diff finds the playlists to be saved
+                $newPlaylists = array_diff($newPlaylists, $existingPlaylists);
+
+                // ... and saves them.
+                foreach ($newPlaylists as $newPlaylist) {
+                    $playlistAndRegionToSave = new PlaylistScreenRegion();
+                    $playlist = $this->playlistRepository->findOneBy(['id' => $newPlaylist]);
+
+                    if (is_null($playlist)) {
+                        throw new InvalidArgumentException('Unknown playlist resource');
+                    }
+
+                    $playlistWeight = array_filter($regionAndPlaylists['playlists'],
+                        /**
+                         * @param array<mixed> $playlistAndWeight
+                         *
+                         * @return bool
+                         */
+                        fn ($playlistAndWeight) => Ulid::fromString($playlistAndWeight['id']) == $playlist->getId());
+
+                    $playlistAndRegionToSave->setPlaylist($playlist);
+                    $playlistAndRegionToSave->setRegion($region);
+                    $playlistAndRegionToSave->setWeight($playlistWeight[0]['weight']);
```

Comment:
can $playlistWeight have a length of 0 ?

Reply:
no. but i can create a length check before setting weight and alternatively set the weight to 0?